src/Integrations/POSCatalog/AsyncGenerator.php

```diff
@@ -79,38 +74,54 @@ public function init( POSIntegration $integration ) {
 	 */
 	public function register_hooks(): void {
 		add_action( self::FEED_GENERATION_ACTION, [ $this, 'feed_generation_action' ] );
-		add_action( self::FEED_DELETION_ACTION, [ $this, 'feed_deletion_action' ] );
+		add_action( self::FEED_DELETION_ACTION, [ $this, 'feed_deletion_action' ], 10, 2 );
 	}
 
 	/**
 	 * Returns the current feed generation status.
 	 * Initiates one if not already running.
 	 *
-	 * @return array The feed generation status.
+	 * @param array|null $args The arguments to pass to the action.
+	 * @return array           The feed generation status.
 	 */
-	public function get_status(): array {
-		$status = get_option( self::OPTION_KEY );
+	public function get_status( ?array $args = null ): array {
+		// Determine the option key based on the integration ID and arguments.
+		$option_key = $this->get_option_key( $args );
+
+		$status = get_option( $option_key );
 
 		if ( false === $status ) {
 			// Clear all previous actions to avoid race conditions.
-			as_unschedule_all_actions( self::FEED_GENERATION_ACTION );
-
-			// Add a bit of delay to avoid race conditions.
-			$delay     = 10;
-			$action_id = as_schedule_single_action( time() + $delay, self::FEED_GENERATION_ACTION, [] );
+			as_unschedule_all_actions( self::FEED_GENERATION_ACTION, [ $option_key ], 'wpfoai' );
 
 			$status = [
-				'action_id' => $action_id,
 				'state'     => self::STATE_SCHEDULED,
 				'progress'  => 0,
 				'processed' => 0,
 				'total'     => -1,
+				'args'      => $args ?? [],
 			];
 
 			update_option(
-				self::OPTION_KEY,
+				$option_key,
 				$status
 			);
+
+			// Start an immediate async action to generate the feed.
+			as_enqueue_async_action(
+				self::FEED_GENERATION_ACTION,
+				[ $option_key ],
+				'wpfoai',
+				true,
+				1
+			);
+
+			// Manually force an async request to be dispatched to process the action immediately.
+			if ( class_exists( ActionScheduler_AsyncRequest_QueueRunner::class ) && class_exists( ActionScheduler_Store::class ) ) {
+				$store         = ActionScheduler_Store::instance();
+				$async_request = new ActionScheduler_AsyncRequest_QueueRunner( $store );
+				$async_request->dispatch();
+			}
 		}
 
 		return $status;
```

Comment:
non-blocking: before returning an existing status with `completed` state, do you think it's worth checking if the file still exists from the status' `path`? If the file no longer exists (like deleted by some admin user accidentally or some plugin), it regenerates the feed again.